SYCL/ESIMD/api/esimd_bit_ops.cpp

```diff
@@ -0,0 +1,146 @@
+//==------- esimd_bit_ops.cpp  - DPC++ ESIMD on-device test ----------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: gpu
+// UNSUPPORTED: cuda
+// RUN: %clangxx -fsycl %s -fsycl-device-code-split=per_kernel -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+//
+// Smoke test for esimd bit operations APIs.
+
+#include "../esimd_test_utils.hpp"
+
+#include <CL/sycl.hpp>
+#include <sycl/ext/intel/experimental/esimd.hpp>
+#include <iostream>
+
+using namespace cl::sycl;
+using namespace sycl::ext::intel::experimental::esimd;
+
+
+struct bit_op {
+  enum { cbit, fbl, fbh, num_ops };
+};
+
+template <typename T, int N, int Op> struct test_id;
+
+template <typename T> struct char_to_int {
+  using type = typename std::conditional<
+      sizeof(T) == 1,
+      typename std::conditional<std::is_signed<T>::value, int, unsigned>::type,
+      T>::type;
+};
+
+// The main test routine.
+template <typename T, int N, int Op> bool test(queue q) {
```

Comment:
Maybe pick a more verbose name for `N`?
```suggestion
template <typename T, int Nbits, int Op> bool test(queue q) {
```

Reply:
N is not Nbits, it is the number of elements in ESIMD vector, as used in lots of other places.